tests/test_cli.py

```diff
@@ -182,3 +191,169 @@ def test_find_designated_time_function(self):
+        """ symlink must be placed at the address that is specified from the input arguments """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-l',
+            '-d', "/dst_folder/"
+        ]
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.get_or_create(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE)
+        with mock.patch('os.symlink') as mock_symlink:
+            cli_copy.main()
+        self.assertEqual(
+            [call(dst='/dst_folder/testing_file.test', src='/testing_folder/testing_file.test'),
+             call(dst='/dst_folder/new_loc_add', src='/new_loc_add')],
+            mock_symlink.call_args_list)
+        for dataset in Dataset.objects.all():
+            dataset.dataseturi_set.filter(
+                uri='/new_loc_add', dataset=dataset, service=LOCAL_FILE_SERVICE).delete()
+
+    @mock.patch('os.symlink')
+    @mock.patch('geospaas_processing.cli.copy.path')
+    def test_correct_content_of_flag_file(self, mock_isfile, mock_link):
+        """ flag file should contain this 'type: test_type' information """
+        mock_isfile.isfile.return_value = True
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l',
+            '-t', 'test_type'
+        ]
+        temp_directory = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory.name)
+        cli_copy.main()
+        fd = os.open(temp_directory.name + '/testing_file.test.flag', os.O_RDONLY)
+        self.assertEqual(os.read(fd, 200),
+                         b"type: test_type\nurl: https://scihub.copernicus.eu/apihub/odata/v1/Produ"
+                         + b"cts('6127111d-c9bd-4689-bab5-412dd39e1e81')/$value\nurl: https://scihu"
+                         + b"b.copernicus.eu/the_second_fakeurl\n")
+
+    def test_copying_accomplishment_and_consistency(self):
+        """ A symlink of actual sample file should be copied to the destination folder """
+        sys.argv = [
+            "",
+            '-b', "2018-04-01",
+            '-e', "2018-04-09",
+            '-f', '-l'
+        ]
+        temp_directory_dst = tempfile.TemporaryDirectory()
+        sys.argv.append('-d')
+        sys.argv.append(temp_directory_dst.name)
+        os.mkdir("/testing_folder")
+        Path("/testing_folder/testing_file.test").touch()
+        cli_copy.main()
+        self.assertTrue(os.path.islink(temp_directory_dst.name+'/testing_file.test'))
+        cli_copy.main()  # assertion of completion of copy for the second time without error
+        self.assertTrue(os.path.islink(temp_directory_dst.name+'/testing_file.test'))
+        shutil.rmtree("/testing_folder")
+
+    def test_copying_accomplishment_and_consistency(self):
```

Comment:
This test method has the same name as the previous one, so the first one is never executed.

Reply:
Thanks for scrutinising comment! I have tested them with other names! I forget to change the name!